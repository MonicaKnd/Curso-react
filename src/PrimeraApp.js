import React from 'react';
import PropTypes from 'prop-types';
// import React, { Fragment } from 'react';

// funtion component
const PrimeraApp = ({saludo, subtitulo }) => {

    // if (!saludo){
    //     throw new Error ('El saludo es necesario');
    // }

    // const saludo = 9000
    // console.log(props);
    // {
    //     nombre: 'Moni',
    //     edad: 45
    // }

    return (
        <>
            <h1>{saludo}</h1>
            <pre>{/*JSON.stringify(saludo, null, 3)*/}</pre>
            <p>{subtitulo}</p>
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'Este es el subtitulo'
}

export default PrimeraApp;