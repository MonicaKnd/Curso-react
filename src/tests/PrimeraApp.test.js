import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';

// import { render } from '@testing-library/react';
import PrimeraApp from "../PrimeraApp";

describe('Pruebas en <PrimeraApp />', () => {
    // test('debe de mostrar el mensaje "Hola, soy Guku', () => {
        
    //     const saludo = 'Hola, soy goku';
    //     const { getByText } = render( <PrimeraApp saludo={ saludo } /> );

    //     expect( getByText(saludo) ).toBeInTheDocument();
        
    // })

    test('debe mostrar <PrimeraApp /> correctamente', () => {
        const saludo = 'Hola picoro';
        const wrapper = shallow(<PrimeraApp  saludo={saludo} /> )
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar el subtitulo enviado por props', () => {
        const saludo = 'Hola picoro';
        const subtitulo = 'Este es el subtitulo';
        const wrapper = shallow(
            <PrimeraApp  
                saludo={saludo} 
                subtitulo={subtitulo}
            /> 
        );

        const textoParrafo = wrapper.find('p').text();
        expect( textoParrafo).toBe(subtitulo);
    })
    
    
    
})
