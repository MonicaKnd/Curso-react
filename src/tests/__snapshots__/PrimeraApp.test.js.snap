// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola picoro
  </h1>
  <pre />
  <p>
    Este es el subtitulo
  </p>
</Fragment>
`;
